ui/src/main/java/org/pentaho/di/ui/trans/steps/pentahoreporting/PentahoReportingOutputDialog.java

```diff
@@ -154,34 +163,95 @@ public void widgetSelected( SelectionEvent e ) {
+    wInputField.addModifyListener( lsMod );
+    FormData fdInputField = new FormData();
+    fdInputField.left = new FormAttachment( middle, 0 );
+    fdInputField.top = new FormAttachment( wStepname, margin + 5 );
+    fdInputField.right = new FormAttachment( 100, 0 );
+    wInputField.setLayoutData( fdInputField );
+
+    String[] fieldNames = new String[] {};
+    try {
+      fieldNames = Const.sortStrings( transMeta.getPrevStepFields( stepMeta ).getFieldNames() );
+    } catch ( KettleException e ) {
+      log.logError( "Unexpected error getting fields from previous steps...", e );
+    }
+
+    wInputField.setItems( fieldNames );
+
+    // output file field line
+    wlOutputField = new Label( shell, SWT.RIGHT );
+    wlOutputField.setText( BaseMessages.getString( PKG, "PentahoReportingOutputDialog.OutputFilenameField.Label" ) );
+    props.setLook( wlOutputField );
+    FormData fdlOutputField = new FormData();
+    fdlOutputField.left = new FormAttachment( 0, 0 );
+    fdlOutputField.top = new FormAttachment( wInputField, margin + 5 );
+    fdlOutputField.right = new FormAttachment( middle, -margin );
+    wlOutputField.setLayoutData( fdlOutputField );
+    wOutputField = new Combo( shell, SWT.SINGLE | SWT.LEFT | SWT.BORDER );
+    props.setLook( wOutputField );
+    wOutputField.addModifyListener( lsMod );
+    FormData fdOutputField = new FormData();
+    fdOutputField.left = new FormAttachment( middle, 0 );
+    fdOutputField.top = new FormAttachment( wInputField, margin + 5 );
+    fdOutputField.right = new FormAttachment( 100, 0 );
+    wOutputField.setLayoutData( fdOutputField );
+    wOutputField.setItems( fieldNames );
+
+    //Use values from fields
+    wlUseValuesFromFields = new Label( shell, SWT.RIGHT );
+    wlUseValuesFromFields.setText( BaseMessages.getString( PKG, "PentahoReportingOutputDialog.UseValuesFromField.Label" ) );
+    props.setLook( wlUseValuesFromFields );
+    FormData fdlUseValuesFromFields = new FormData();
+    fdlUseValuesFromFields.left = new FormAttachment( 10, 10 );
+    fdlUseValuesFromFields.top = new FormAttachment( wOutputField, margin + 15 );
+    fdlUseValuesFromFields.right = new FormAttachment( middle, -margin );
+    wlUseValuesFromFields.setLayoutData( fdlUseValuesFromFields );
+    wUseValuesFromFields = new Button( shell, SWT.CHECK );
+    props.setLook( wUseValuesFromFields );
+    wUseValuesFromFields.addSelectionListener( selMod );
+    FormData fdUseValuesFromFields = new FormData();
+    fdUseValuesFromFields.left = new FormAttachment( middle, 10 );
+    fdUseValuesFromFields.top = new FormAttachment( wOutputField, margin + 15 );
+    fdUseValuesFromFields.right = new FormAttachment( 100, 10 );
+    wUseValuesFromFields.setLayoutData( fdUseValuesFromFields );
+    wUseValuesFromFields.setSelection( input.getUseValuesFromFields() );
+    wUseValuesFromFields.addSelectionListener( new SelectionAdapter() {
+      public void widgetSelected( SelectionEvent e ) {
```

Comment:
Please add the "@Override" annotation in here

Reply:
Done